stock_cycle_count/reports/report_stock_location_accuracy.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2017 ForgeFlow S.L.
+#   (http://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class LocationAccuracyReport(models.AbstractModel):
+    _name = "report.stock_location_accuracy"
+    _description = "Location Accuracy Report"
+
+    @api.model
+    def _get_inventory_domain(self, loc_id, exclude_sublocation=True):
+        return [
+            ("location_id", "=", loc_id),
+            ("exclude_sublocation", "=", exclude_sublocation),
+            ("filter", "=", "none"),
+            ("state", "=", "done"),
+        ]
+
+    @api.model
+    def _get_location_data(self, locations):
+        data = dict()
+        inventory_obj = self.env["stock.inventory"]
+        for loc in locations:
+            counts = inventory_obj.search(self._get_inventory_domain(loc.id))
```

Comment:
You should change the search to use multi recordset, call search() once, then split per location.